### ISSUE TITLE
[Android] Virtual display - Surface view blank on App Resume

### ISSUE THREAD

**Issue by Neelansh-ns** [flutter/flutter#151178]
### Steps to reproduce

This is a duplicate of #144219 

This issue was resolved earlier when tested on the Flutter 3.19.6 version.

This is again happening on the flutter stable 3.22.0 and the latest master as well.

1. Run code sample on Android device (Tested on Redmi Note 9 Pro Max)
2. Leave the app and come back.
3. This can be to the home screen or any other app, it doesn't matter.
4. You should now only see the color of the container behind the platform view.

### Expected results

The video should continue to play without the screen going black.

### Actual results

The AndroidView is blank on app resume.
The debug console is spammed with

```
E/BufferQueueProducer(28102): [ImageReader-1080x2000f22m4-28102-0](id:6dc600000002,api:1,p:1104,c:28102) dequeueBuffer: BufferQueue has been abandoned 
```

### Code sample

(basic surfaceview)
https://github.com/timbotimbo/flutter-GLTest/tree/initial

(basic surfaceview + additional surfaceholder)
https://github.com/timbotimbo/flutter-GLTest/tree/master

### Screenshots or Video

<details open>
<summary>Screenshots / Video demonstration</summary>

[Upload media here]

</details>

### Logs

<details open><summary>Logs</summary>

```console
2024-07-02 22:31:36.560  2920-3296  ActivityManagerWrapper  com.miui.home                        E  getRecentTasks: taskId=39285   userId=0   baseIntent=Intent { act=android.intent.action.MAIN flag=805306368 cmp=ComponentInfo{com.example.video_player_surface_example/com.example.video_player_surface_example.MainActivity} }
2024-07-02 22:31:36.913  2920-3296  ActivityManagerWrapper  com.miui.home                        E  getRecentTasks: taskId=39285   userId=0   baseIntent=Intent { act=android.intent.action.MAIN flag=805306368 cmp=ComponentInfo{com.example.video_player_surface_example/com.example.video_player_surface_example.MainActivity} }
2024-07-02 22:31:36.996  2920-2982  InstallApkReceiver      com.miui.home                        E  uninstall:package:com.example.video_player_surface_example
2024-07-02 22:31:37.069  2920-2982  InstallApkReceiver      com.miui.home                        E  install:package:com.example.video_player_surface_example
2024-07-02 22:31:37.238  1468-2450  AppScanObserverService  system_server                        E  Try to add a invalid package: com.example.video_player_surface_example | 10105 | 0
2024-07-02 22:31:37.291  1468-2060  VerityUtils             system_server                        E  Failed to measure fs-verity, errno 1: /data/app/~~AFgIZfEnPXPwV4Zc0M9t6Q==/com.example.video_player_surface_example-Ygd2on25qek0ANSnZb-xMQ==/base.apk
2024-07-02 22:31:37.631  1468-2060  VerityUtils             system_server                        E  Failed to measure fs-verity, errno 1: /data/app/~~AFgIZfEnPXPwV4Zc0M9t6Q==/com.example.video_player_surface_example-Ygd2on25qek0ANSnZb-xMQ==/base.apk
2024-07-02 22:31:37.647  2920-3296  ActivityManagerWrapper  com.miui.home                        E  getRecentTasks: taskId=39287   userId=0   baseIntent=Intent { act=android.intent.action.MAIN flag=805306368 cmp=ComponentInfo{com.example.video_player_surface_example/com.example.video_player_surface_example.MainActivity} }
---------------------------- PROCESS STARTED (942) for package com.example.video_player_surface_example ----------------------------
2024-07-02 22:31:37.793   942-942   Compatibil...geReporter com....video_player_surface_example  D  Compat change id reported: 171979766; UID 10105; state: ENABLED
2024-07-02 22:31:37.803   942-977   AppScoutStateMachine    com....video_player_surface_example  D  942-ScoutStateMachinecreated
2024-07-02 22:31:37.822   942-942   ziparchive              com....video_player_surface_example  W  Unable to open '/data/app/~~AFgIZfEnPXPwV4Zc0M9t6Q==/com.example.video_player_surface_example-Ygd2on25qek0ANSnZb-xMQ==/base.dm': No such file or directory
2024-07-02 22:31:37.822   942-942   ziparchive              com....video_player_surface_example  W  Unable to open '/data/app/~~AFgIZfEnPXPwV4Zc0M9t6Q==/com.example.video_player_surface_example-Ygd2on25qek0ANSnZb-xMQ==/base.dm': No such file or directory
2024-07-02 22:31:37.984   942-942   nativeloader            com....video_player_surface_example  D  Configuring clns-4 for other apk /data/app/~~AFgIZfEnPXPwV4Zc0M9t6Q==/com.example.video_player_surface_example-Ygd2on25qek0ANSnZb-xMQ==/base.apk. target_sdk_version=34, uses_libraries=, library_path=/data/app/~~AFgIZfEnPXPwV4Zc0M9t6Q==/com.example.video_player_surface_example-Ygd2on25qek0ANSnZb-xMQ==/lib/arm64:/data/app/~~AFgIZfEnPXPwV4Zc0M9t6Q==/com.example.video_player_surface_example-Ygd2on25qek0ANSnZb-xMQ==/base.apk!/lib/arm64-v8a, permitted_path=/data:/mnt/expand:/data/user/0/com.example.video_player_surface_example
2024-07-02 22:31:38.002   942-942   Perf                    com....video_player_surface_example  I  Connecting to perf service.
2024-07-02 22:31:38.021   942-942   GraphicsEnvironment     com....video_player_surface_example  V  ANGLE Developer option for 'com.example.video_player_surface_example' set to: 'default'
2024-07-02 22:31:38.022   942-942   GraphicsEnvironment     com....video_player_surface_example  V  Neither updatable production driver nor prerelease driver is supported.
2024-07-02 22:31:38.030   942-942   ForceDarkHelperStubImpl com....video_player_surface_example  I  initialize for com.example.video_player_surface_example , ForceDarkOrigin
2024-07-02 22:31:38.032   942-942   OpenGLRenderer          com....video_player_surface_example  D  JNI_OnLoad success
2024-07-02 22:31:38.032   942-942   MiuiForceDarkConfig     com....video_player_surface_example  I  setConfig density:2.750000, mainRule:0, secondaryRule:0, tertiaryRule:0
2024-07-02 22:31:38.035   942-942   NetworkSecurityConfig   com....video_player_surface_example  D  No Network Security Config specified, using platform default
2024-07-02 22:31:38.036   942-942   NetworkSecurityConfig   com....video_player_surface_example  D  No Network Security Config specified, using platform default
2024-07-02 22:31:38.178   942-1079  ResourceExtractor       com....video_player_surface_example  I  Found extracted resources res_timestamp-1-1719939030387
2024-07-02 22:31:38.178   942-1079  ResourceExtractor       com....video_player_surface_example  I  Resource version mismatch res_timestamp-1-1719939696786
2024-07-02 22:31:38.451   942-1079  ResourceExtractor       com....video_player_surface_example  I  Extracted baseline resource assets/flutter_assets/kernel_blob.bin
2024-07-02 22:31:38.452   942-1079  ResourceExtractor       com....video_player_surface_example  I  Extracted baseline resource assets/flutter_assets/vm_snapshot_data
2024-07-02 22:31:38.525   942-1079  ResourceExtractor       com....video_player_surface_example  I  Extracted baseline resource assets/flutter_assets/isolate_snapshot_data
2024-07-02 22:31:38.581   942-942   AdrenoGLES-0            com....video_player_surface_example  I  QUALCOMM build                   : 820d4a4028, I5580630f49
                                                                                                    Build Date                       : 11/04/21
                                                                                                    OpenGL ES Shader Compiler Version: EV031.32.02.16
                                                                                                    Local Branch                     : 
                                                                                                    Remote Branch                    : 
                                                                                                    Remote Branch                    : 
                                                                                                    Reconstruct Branch               : 
2024-07-02 22:31:38.581   942-942   AdrenoGLES-0            com....video_player_surface_example  I  Build Config                     : S P 10.0.7 AArch64
2024-07-02 22:31:38.581   942-942   AdrenoGLES-0            com....video_player_surface_example  I  Driver Path                      : /vendor/lib64/egl/libGLESv2_adreno.so
2024-07-02 22:31:38.601   942-942   AdrenoGLES-0            com....video_player_surface_example  I  PFP: 0x016ee190, ME: 0x00000000
2024-07-02 22:31:38.669   942-942   ViewScrollStubImpl      com....video_player_surface_example  D  the value of mScrollFrictionRes is: 0.015
2024-07-02 22:31:38.739   942-942   surface_exampl          com....video_player_surface_example  W  Accessing hidden method Landroid/view/accessibility/AccessibilityNodeInfo;->getSourceNodeId()J (unsupported,test-api, reflection, allowed)
2024-07-02 22:31:38.739   942-942   surface_exampl          com....video_player_surface_example  W  Accessing hidden method Landroid/view/accessibility/AccessibilityRecord;->getSourceNodeId()J (unsupported, reflection, allowed)
2024-07-02 22:31:38.739   942-942   surface_exampl          com....video_player_surface_example  W  Accessing hidden field Landroid/view/accessibility/AccessibilityNodeInfo;->mChildNodeIds:Landroid/util/LongArray; (unsupported, reflection, allowed)
2024-07-02 22:31:38.739   942-942   surface_exampl          com....video_player_surface_example  W  Accessing hidden method Landroid/util/LongArray;->get(I)J (unsupported, reflection, allowed)
2024-07-02 22:31:38.774   942-1184  flutter                 com....video_player_surface_example  I  The Dart VM service is listening on http://127.0.0.1:41347/_7Xo2JuQ0iw=/
2024-07-02 22:31:38.786   942-942   MiuiMultiWindowUtils    com....video_player_surface_example  D  freeform resolution args raw data:{  "zizhan":{    "freeform_args": {      "inner":{        "vertical_portrait":{"aspect_ratio":0.626, "original_ratio":0.5643,"original_scale":0.74,"top_margin":0.168,"left_margin":0.484},        "horizontal_portrait":{"aspect_ratio":0.626, "original_ratio":0.5643,"original_scale":0.74,"top_margin":0.1222,"left_margin":0.59745},        "vertical_landscape":{"aspect_ratio":1.6, "original_ratio":1,"original_scale":0.604,"top_margin":0.2596,"left_margin":0.2624},        "horizontal_landscape":{"aspect_ratio":1.6, "original_ratio":1,"original_scale":0.604,"top_margin":0.213,"left_margin":0.3758}      },      "outer":{        "vertical_portrait":{"aspect_ratio":0.626, "original_ratio":1,"original_scale":0.74,"top_margin":0.0753,"left_margin":-1},        "horizontal_portrait":{"aspect_ratio":0.626, "original_ratio":1,"original_scale":0.5756,"top_margin":-1,"left_margin":0.0753},        "vertical_landscape":{"aspect_ratio":1.6, "original_ratio":0.6847,"original_scale":0.587,"top_margin":0.0753,"left_margin":-1},        "horizontal_landscape":{"aspect_ratio":1.6, "original_ratio":0.6847,"original_scale":0.587,"top_margin":-1,"left_margin":0.0753}}    },    "mini_freeform_args":{      "inner":{        "vertical_portrait":{"aspect_ratio":0.626, "original_ratio":0.147},        "horizontal_portrait":{"aspect_ratio":0.626, "original_ratio":0.147},        "vertical_landscape":{"aspect_ratio":1.6, "original_ratio":0.165},        "horizontal_landscape":{"aspect_ratio":1.6, "original_ratio":0.165}      },      "outer":{        "vertical_portrait":{"aspect_ratio":0.626, "original_ratio":0.26},        "horizontal_portrait":{"aspect_ratio":0.626, "original_ratio":0.26},        "vertical_landscape":{"aspect_ratio":1.6, "original_ratio":0.293},        "horizontal_landscape":{"aspect_ratio":1.6, "original_ratio":0.293}}    }  },  "wide_default":{    "freeform_args": {        "vertical_portrait":{"aspect_ratio":0.625, "original_ratio":0.58,"original_scale":0.7574,"top_margin":0.125,"left_margin":-1},        "horizontal_portrait":{"aspect_ratio":0.625, "original_ratio":0.58,"original_scale":0.7574,"top_margin":-1,"left_margin":0.0714},        "vertical_landscape":{"aspect_ratio":1.6, "original_ratio":1,"original_scale":0.5153,"top_margin":0.125,"left_margin":-1},        "horizontal_landscape":{"aspect_ratio":1.6, "original_ratio":1,"original_scale":0.5153,"top_margin":-1,"left_margin":0.0714}    },    "mini_freeform_args":{        "vertical_portrait":{"aspect_ratio":0.5625, "original_ratio":0.144},        "horizontal_portrait":{"aspect_ratio":0.5625, "original_ratio":0.144},        "vertical_landscape":{"aspect_ratio":-1, "original_ratio":0.144},        "horizontal_landscape":{"aspect_ratio":-1, "original_ratio":0.144}    }  },  "narrow_default": {    "freeform_args": {        "vertical_portrait":{"aspect_ratio":0.5625, "original_ratio":1,"original_scale":0.7182,"top_margin":0.142,"left_margin":-1},        "horizontal_portrait":{"aspect_ratio":0.8, "original_ratio":1,"original_scale":0.7182,"top_margin":-1,"left_margin":0.05},        "vertical_landscape":{"aspect_ratio":1.6, "original_ratio":0.6545,"original_scale":0.4473,"top_margin":0.142,"left_margin":-1},        "horizontal_landscape":{"aspect_ratio":1.6, "original_ratio":0.6545,"original_scale":0.4473,"top_margin":-1,"left_margin":0.05}    },    "mini_freeform_args":{        "vertical_portrait":{"aspect_ratio":0.5625, "original_ratio":0.3},        "horizontal_portrait":{"aspect_ratio":0.5625, "original_ratio":0.3},        "vertical_landscape":{"aspect_ratio":1.7778, "original_ratio":0.3},        "horizontal_landscape":{"aspect_ratio":1.7778, "original_ratio":0.3}    }  },  "regular_default": {    "freeform_args": {      "vertical_portrait":{"aspect_ratio":0.625, "original_ratio":1,"original_scale":0.7,"top_margin":0.109,"left_margin":-1},      "horizontal_portrait":{"aspect_ratio":0.6667, "original_ratio":1,"original_scale":0.6102,"top_margin":-1,"left_margin":0.026},      "vertical_landscape":{"aspect_ratio
2024-07-02 22:31:38.798   942-942   System.err              com....video_player_surface_example  W  org.json.JSONException: No value for excalibur
2024-07-02 22:31:38.803   942-942   System.err              com....video_player_surface_example  W  	at org.json.JSONObject.get(JSONObject.java:398)
2024-07-02 22:31:38.803   942-942   System.err              com....video_player_surface_example  W  	at org.json.JSONObject.getJSONObject(JSONObject.java:618)
2024-07-02 22:31:38.804   942-942   System.err              com....video_player_surface_example  W  	at android.util.MiuiMultiWindowUtils.initFreeFormResolutionArgsOfDevice(MiuiMultiWindowUtils.java:1433)
2024-07-02 22:31:38.804   942-942   System.err              com....video_player_surface_example  W  	at android.util.MiuiMultiWindowUtils.initFreeFormResolutionArgs(MiuiMultiWindowUtils.java:1427)
2024-07-02 22:31:38.804   942-942   System.err              com....video_player_surface_example  W  	at android.util.MiuiMultiWindowUtils.<clinit>(MiuiMultiWindowUtils.java:209)
2024-07-02 22:31:38.804   942-942   System.err              com....video_player_surface_example  W  	at com.android.internal.policy.DecorViewMultiWinStubImpl.<init>(DecorViewMultiWinStubImpl.java:19)
2024-07-02 22:31:38.804   942-942   System.err              com....video_player_surface_example  W  	at com.android.internal.policy.DecorViewMultiWinStubImpl$Provider$SINGLETON.<clinit>(DecorViewMultiWinStubImpl$Provider.java:21)
2024-07-02 22:31:38.804   942-942   System.err              com....video_player_surface_example  W  	at com.android.internal.policy.DecorViewMultiWinStubImpl$Provider.provideSingleton(DecorViewMultiWinStubImpl$Provider.java:10)
2024-07-02 22:31:38.804   942-942   System.err              com....video_player_surface_example  W  	at com.android.internal.policy.DecorViewMultiWinStubImpl$Provider.provideSingleton(DecorViewMultiWinStubImpl$Provider.java:8)
2024-07-02 22:31:38.804   942-942   System.err              com....video_player_surface_example  W  	at com.miui.base.MiuiStubRegistry.get(MiuiStubRegistry.java:129)
2024-07-02 22:31:38.804   942-942   System.err              com....video_player_surface_example  W  	at com.miui.base.MiuiStubRegistry.get(MiuiStubRegistry.java:123)
2024-07-02 22:31:38.804   942-942   System.err              com....video_player_surface_example  W  	at com.miui.base.MiuiStubUtil.getInstance(MiuiStubUtil.java:46)
2024-07-02 22:31:38.804   942-942   System.err              com....video_player_surface_example  W  	at com.miui.base.MiuiStubUtil.getImpl(MiuiStubUtil.java:30)
2024-07-02 22:31:38.805   942-942   System.err              com....video_player_surface_example  W  	at com.android.internal.policy.DecorViewMultiWinStub$SingletonHolder.<clinit>(DecorViewMultiWinStub.java:11)
2024-07-02 22:31:38.805   942-942   System.err              com....video_player_surface_example  W  	at com.android.internal.policy.DecorViewMultiWinStub.newInstance(DecorViewMultiWinStub.java:16)
2024-07-02 22:31:38.805   942-942   System.err              com....video_player_surface_example  W  	at com.android.internal.policy.DecorView.<init>(DecorView.java:335)
2024-07-02 22:31:38.805   942-942   System.err              com....video_player_surface_example  W  	at com.android.internal.policy.PhoneWindow.generateDecor(PhoneWindow.java:2380)
2024-07-02 22:31:38.805   942-942   System.err              com....video_player_surface_example  W  	at com.android.internal.policy.PhoneWindow.installDecor(PhoneWindow.java:2760)
2024-07-02 22:31:38.805   942-942   System.err              com....video_player_surface_example  W  	at com.android.internal.policy.PhoneWindow.setContentView(PhoneWindow.java:487)
2024-07-02 22:31:38.805   942-942   System.err              com....video_player_surface_example  W  	at com.android.internal.policy.PhoneWindow.setContentView(PhoneWindow.java:478)
2024-07-02 22:31:38.805   942-942   System.err              com....video_player_surface_example  W  	at android.app.Activity.setContentView(Activity.java:3618)
2024-07-02 22:31:38.805   942-942   System.err              com....video_player_surface_example  W  	at io.flutter.embedding.android.FlutterActivity.onCreate(FlutterActivity.java:645)
2024-07-02 22:31:38.805   942-942   System.err              com....video_player_surface_example  W  	at android.app.Activity.performCreate(Activity.java:8163)
2024-07-02 22:31:38.805   942-942   System.err              com....video_player_surface_example  W  	at android.app.Activity.performCreate(Activity.java:8130)
2024-07-02 22:31:38.805   942-942   System.err              com....video_player_surface_example  W  	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1330)
2024-07-02 22:31:38.805   942-942   System.err              com....video_player_surface_example  W  	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3744)
2024-07-02 22:31:38.806   942-942   System.err              com....video_player_surface_example  W  	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3938)
2024-07-02 22:31:38.806   942-942   System.err              com....video_player_surface_example  W  	at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:103)
2024-07-02 22:31:38.806   942-942   System.err              com....video_player_surface_example  W  	at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
2024-07-02 22:31:38.806   942-942   System.err              com....video_player_surface_example  W  	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
2024-07-02 22:31:38.806   942-942   System.err              com....video_player_surface_example  W  	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2291)
2024-07-02 22:31:38.806   942-942   System.err              com....video_player_surface_example  W  	at android.os.Handler.dispatchMessage(Handler.java:106)
2024-07-02 22:31:38.806   942-942   System.err              com....video_player_surface_example  W  	at android.os.Looper.loopOnce(Looper.java:210)
2024-07-02 22:31:38.806   942-942   System.err              com....video_player_surface_example  W  	at android.os.Looper.loop(Looper.java:299)
2024-07-02 22:31:38.806   942-942   System.err              com....video_player_surface_example  W  	at android.app.ActivityThread.main(ActivityThread.java:8319)
2024-07-02 22:31:38.806   942-942   System.err              com....video_player_surface_example  W  	at java.lang.reflect.Method.invoke(Native Method)
2024-07-02 22:31:38.806   942-942   System.err              com....video_player_surface_example  W  	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:556)
2024-07-02 22:31:38.806   942-942   System.err              com....video_player_surface_example  W  	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1038)
2024-07-02 22:31:38.807   942-942   IS_CTS_MODE             com....video_player_surface_example  D  false
2024-07-02 22:31:38.807   942-942   MULTI_WIND...CH_ENABLED com....video_player_surface_example  D  false
2024-07-02 22:31:38.808   942-942   IS_CTS_MODE             com....video_player_surface_example  D  false
2024-07-02 22:31:38.808   942-942   MULTI_WIND...CH_ENABLED com....video_player_surface_example  D  false
2024-07-02 22:31:38.808   942-942   IS_CTS_MODE             com....video_player_surface_example  D  false
2024-07-02 22:31:38.808   942-942   MULTI_WIND...CH_ENABLED com....video_player_surface_example  D  false
2024-07-02 22:31:38.815   942-942   DecorView[]             com....video_player_surface_example  D  getWindowModeFromSystem  windowmode is 1
2024-07-02 22:31:38.829   942-942   Activity                com....video_player_surface_example  W  PerfMonitor: Slow Operation: Activity com.example.video_player_surface_example/.MainActivity onCreate took 716ms
2024-07-02 22:31:38.884   942-942   DecorView[]             com....video_player_surface_example  D  getWindowModeFromSystem  windowmode is 1
2024-07-02 22:31:38.885   942-942   DecorView[]             com....video_player_surface_example  D  updateDecorCaptionStatus displayWindowDecor is false
2024-07-02 22:31:38.904   942-942   Looper                  com....video_player_surface_example  W  PerfMonitor looperActivity : package=com.example.video_player_surface_example/.MainActivity time=1ms latency=1106ms running=4ms  procState=-1  historyMsgCount=4 (msgIndex=1 wall=831ms seq=4 running=270ms runnable=41ms io=54ms late=275ms h=android.app.ActivityThread$H w=159) (msgIndex=2 wall=831ms seq=4 running=270ms runnable=41ms io=54ms late=275ms h=android.app.ActivityThread$H w=159) (msgIndex=3 wall=278ms seq=3 running=185ms runnable=49ms io=14ms h=android.app.ActivityThread$H w=110) (msgIndex=4 wall=831ms seq=4 running=270ms runnable=41ms io=54ms late=275ms h=android.app.ActivityThread$H w=159)
2024-07-02 22:31:38.906   942-942   Choreographer           com....video_player_surface_example  I  Skipped 50 frames!  The application may be doing too much work on its main thread.
2024-07-02 22:31:39.185   942-942   SurfaceView             com....video_player_surface_example  D  UPDATE null, mIsCastMode = false
2024-07-02 22:31:39.187   942-1000  hw-ProcessState         com....video_player_surface_example  D  Binder ioctl to enable oneway spam detection failed: Invalid argument
2024-07-02 22:31:39.202   942-942   Looper                  com....video_player_surface_example  W  PerfMonitor doFrame : time=296ms vsyncFrame=0 latency=850ms procState=-1 historyMsgCount=6 (msgIndex=1 wall=831ms seq=4 running=270ms runnable=41ms io=54ms h=android.app.ActivityThread$H w=159) (msgIndex=2 wall=831ms seq=4 running=270ms runnable=41ms io=54ms h=android.app.ActivityThread$H w=159) (msgIndex=3 wall=296ms seq=7 running=245ms runnable=19ms io=2ms late=850ms h=android.view.Choreographer$FrameHandler c=android.view.Choreographer$FrameDisplayEventReceiver) (msgIndex=4 wall=831ms seq=4 running=270ms runnable=41ms io=54ms h=android.app.ActivityThread$H w=159)
2024-07-02 22:31:39.220   942-942   SurfaceView             com....video_player_surface_example  D  UPDATE Surface(name=SurfaceView[com.example.video_player_surface_example/com.example.video_player_surface_example.MainActivity])/@0x26777b2, mIsProjectionMode = false
2024-07-02 22:31:40.299   942-942   DecorView[]             com....video_player_surface_example  D  getWindowModeFromSystem  windowmode is 1
2024-07-02 22:31:40.299   942-942   DecorView[]             com....video_player_surface_example  D  updateDecorCaptionStatus displayWindowDecor is false
2024-07-02 22:31:40.364   942-1154  surface_exampl          com....video_player_surface_example  E  open libmigui.so failed! dlopen - dlopen failed: library "libmigui.so" not found
2024-07-02 22:31:40.548   942-942   DecorView[]             com....video_player_surface_example  D  onWindowFocusChanged hasWindowFocus true
2024-07-02 22:31:43.998   942-942   PlatformViewsController com....video_player_surface_example  I  Hosting view in a virtual display for platform view: 0
2024-07-02 22:31:44.002   942-942   PlatformViewsController com....video_player_surface_example  I  PlatformView is using SurfaceProducer backend
2024-07-02 22:31:44.113   942-942   IS_CTS_MODE             com....video_player_surface_example  D  false
2024-07-02 22:31:44.113   942-942   MULTI_WIND...CH_ENABLED com....video_player_surface_example  D  false
2024-07-02 22:31:44.148   942-1154  ImageReade...ceProducer com....video_player_surface_example  D  ImageTextureEntry can't wait on the fence on Android < 33
2024-07-02 22:31:44.162   942-942   SurfaceView             com....video_player_surface_example  D  UPDATE null, mIsCastMode = false
2024-07-02 22:31:44.187   942-942   SurfaceView             com....video_player_surface_example  D  UPDATE Surface(name=SurfaceView[])/@0xe620dd6, mIsProjectionMode = false
2024-07-02 22:31:44.640   942-1225  ProfileInstaller        com....video_player_surface_example  D  Installing profile for com.example.video_player_surface_example
2024-07-02 22:31:45.625   942-1223  MediaPlayerNative       com....video_player_surface_example  W  info/warning (3, 0)
2024-07-02 22:31:49.010  1275-1346  Finsky                  com.android.vending                  E  [422] VerifyRequiredSplitTypesInstallTask.mw(653): VerifyApps: Required split types check failed for com.example.video_player_surface_example. (Ask Gemini)
                                                                                                    java.lang.IllegalStateException: Install path is null
                                                                                                    	at com.google.android.finsky.verifier.impl.installtimeverification.VerifyRequiredSplitTypesInstallTask.mw(PG:640)
                                                                                                    	at com.google.android.finsky.verifier.impl.installtimeverification.VerifyInstallTask.mw(PG:41)
                                                                                                    	at akwu.run(PG:228)
                                                                                                    	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
                                                                                                    	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:644)
                                                                                                    	at auxe.run(PG:74)
                                                                                                    	at java.lang.Thread.run(Thread.java:1012)
2024-07-02 22:31:49.048  1275-1346  Finsky                  com.android.vending                  E  [422] VerifyV31SignatureInstallTask.e(159): VerifyApps V31SignatureVerification: Failed to collect certificates for the package: com.example.video_player_surface_example using APK Signature Scheme v3
2024-07-02 22:31:50.662   942-942   BpBinder                com....video_player_surface_example  W  Slow Binder: BpBinder transact took 214 ms, interface=android.app.IActivityClientController, code=24 oneway=false
2024-07-02 22:31:50.683   942-942   DecorView[]             com....video_player_surface_example  D  getWindowModeFromSystem  windowmode is 1
2024-07-02 22:31:50.684   942-942   DecorView[]             com....video_player_surface_example  D  updateDecorCaptionStatus displayWindowDecor is false
2024-07-02 22:31:51.031   942-942   DecorView[]             com....video_player_surface_example  D  updateDecorCaptionStatus displayWindowDecor is false
2024-07-02 22:31:53.250   942-942   DecorView[]             com....video_player_surface_example  D  getWindowModeFromSystem  windowmode is 1
2024-07-02 22:31:53.250   942-942   DecorView[]             com....video_player_surface_example  D  updateDecorCaptionStatus displayWindowDecor is false
2024-07-02 22:31:53.541   942-942   DecorView[]             com....video_player_surface_example  D  updateDecorCaptionStatus displayWindowDecor is false
2024-07-02 22:31:55.689   942-942   DecorView[]             com....video_player_surface_example  D  onWindowFocusChanged hasWindowFocus false
2024-07-02 22:31:56.267  2920-3296  ActivityManagerWrapper  com.miui.home                        E  getRecentTasks: taskId=39287   userId=0   baseIntent=Intent { act=android.intent.action.MAIN flag=805306368 cmp=ComponentInfo{com.example.video_player_surface_example/com.example.video_player_surface_example.MainActivity} }
2024-07-02 22:31:56.271   942-1223  BufferQueueProducer     com....video_player_surface_example  E  [ImageReader-1080x688f22m5-942-2](id:3ae00000006,api:1,p:1128,c:942) dequeueBuffer: BufferQueue has been abandoned
2024-07-02 22:31:56.292   942-981   BufferQueueProducer     com....video_player_surface_example  E  [ImageReader-1080x688f22m5-942-2](id:3ae00000006,api:1,p:1128,c:942) dequeueBuffer: BufferQueue has been abandoned
2024-07-02 22:31:56.336   942-981   BufferQueueProducer     com....video_player_surface_example  E  [ImageReader-1080x688f22m5-942-2](id:3ae00000006,api:1,p:1128,c:942) dequeueBuffer: BufferQueue has been abandoned
2024-07-02 22:31:56.355   942-981   BufferQueueProducer     com....video_player_surface_example  E  [ImageReader-1080x688f22m5-942-2](id:3ae00000006,api:1,p:1128,c:942) dequeueBuffer: BufferQueue has been abandoned
2024-07-02 22:32:00.664   942-1536  BufferQueueProducer     com....video_player_surface_example  E  [ImageReader-1080x688f22m5-942-2](id:3ae00000006,api:1,p:1128,c:942) dequeueBuffer: BufferQueue has been abandoned
2024-07-02 22:32:00.715   942-1536  BufferQueueProducer     com....video_player_surface_example  E  [ImageReader-1080x688f22m5-942-2](id:3ae00000006,api:1,p:1128,c:942) dequeueBuffer: BufferQueue has been abandoned
2024-07-02 22:32:06.541   942-1536  AppScoutStateMachine    com....video_player_surface_example  D  942-ScoutStateMachinecreated
2024-07-02 22:32:06.546   942-1223  BufferQueueProducer     com....video_player_surface_example  E  [ImageReader-1080x688f22m5-942-2](id:3ae00000006,api:1,p:1128,c:942) dequeueBuffer: BufferQueue has been abandoned
2024-07-02 22:32:06.562   942-1536  BufferQueueProducer     com....video_player_surface_example  E  [ImageReader-1080x688f22m5-942-2](id:3ae00000006,api:1,p:1128,c:942) dequeueBuffer: BufferQueue has been abandoned
2024-07-02 22:32:06.574   942-942   SurfaceView             com....video_player_surface_example  D  UPDATE Surface(name=SurfaceView[])/@0xe620dd6, mIsCastMode = false
2024-07-02 22:32:06.575   942-942   SurfaceView             com....video_player_surface_example  D  UPDATE Surface(name=SurfaceView[])/@0xe620dd6, mIsProjectionMode = false
2024-07-02 22:32:06.579   942-1536  BufferQueueProducer     com....video_player_surface_example  E  [ImageReader-1080x688f22m5-942-2](id:3ae00000006,api:1,p:1128,c:942) dequeueBuffer: BufferQueue has been abandoned
2024-07-02 22:32:06.605   942-1536  BufferQueueProducer     com....video_player_surface_example  E  [SurfaceView[]#3(BLAST Consumer)3](id:3ae00000004,api:3,p:1407,c:942) queueBuffer: BufferQueue has been abandoned
2024-07-02 22:32:06.607   942-1536  MediaPlayerNative       com....video_player_surface_example  W  info/warning (805, -19)
2024-07-02 22:32:06.608   942-1536  MediaPlayerNative       com....video_player_surface_example  W  info/warning (805, -38)
2024-07-02 22:32:06.654   942-1536  BufferQueueProducer     com....video_player_surface_example  E  [SurfaceView[]#3(BLAST Consumer)3](id:3ae00000004,api:3,p:1407,c:942) connect: BufferQueue has been abandoned
2024-07-02 22:32:06.714  2920-3296  ActivityManagerWrapper  com.miui.home                        E  getRecentTasks: taskId=39287   userId=0   baseIntent=Intent { act=android.intent.action.MAIN flag=805306368 cmp=ComponentInfo{com.example.video_player_surface_example/com.example.video_player_surface_example.MainActivity} }
2024-07-02 22:32:06.721   942-942   IS_CTS_MODE             com....video_player_surface_example  D  false
2024-07-02 22:32:06.721   942-942   MULTI_WIND...CH_ENABLED com....video_player_surface_example  D  false
2024-07-02 22:32:06.781   942-942   SurfaceView             com....video_player_surface_example  D  UPDATE Surface(name=SurfaceView[com.example.video_player_surface_example/com.example.video_player_surface_example.MainActivity])/@0x4fb6328, mIsProjectionMode = false
2024-07-02 22:32:06.818   942-942   SurfaceView             com....video_player_surface_example  D  UPDATE null, mIsCastMode = false
2024-07-02 22:32:06.838   942-942   SurfaceView             com....video_player_surface_example  D  UPDATE Surface(name=SurfaceView[])/@0x7edd1e6, mIsProjectionMode = false
2024-07-02 22:32:06.840   942-942   DecorView[]             com....video_player_surface_example  D  onWindowFocusChanged hasWindowFocus true
2024-07-02 22:32:12.300   942-942   MIUIInput               com....video_player_surface_example  D  [KeyEvent] ViewRootImpl windowName 'com.example.video_player_surface_example/com.example.video_player_surface_example.MainActivity', KeyEvent { action=ACTION_DOWN, keyCode=KEYCODE_BACK, scanCode=0, metaState=0, flags=0x8, repeatCount=0, eventTime=57149548, downTime=57149548, deviceId=-1, source=0x101, displayId=0 }, phoneEventTime=22:32:12.261
2024-07-02 22:32:12.306   942-942   MIUIInput               com....video_player_surface_example  D  [KeyEvent] ViewRootImpl windowName 'com.example.video_player_surface_example/com.example.video_player_surface_example.MainActivity', KeyEvent { action=ACTION_UP, keyCode=KEYCODE_BACK, scanCode=0, metaState=0, flags=0x8, repeatCount=0, eventTime=57149548, downTime=57149548, deviceId=-1, source=0x101, displayId=0 }, phoneEventTime=22:32:12.261
2024-07-02 22:32:12.662   942-942   SurfaceView             com....video_player_surface_example  D  UPDATE Surface(name=SurfaceView[])/@0x7edd1e6, mIsCastMode = false
2024-07-02 22:32:12.663   942-942   SurfaceView             com....video_player_surface_example  D  UPDATE Surface(name=SurfaceView[])/@0x7edd1e6, mIsProjectionMode = false
2024-07-02 22:32:12.665   942-942   MediaPlayer             com....video_player_surface_example  V  resetDrmState:  mDrmInfo=null mDrmProvisioningThread=null mPrepareDrmInProgress=false mActiveDrmScheme=false
2024-07-02 22:32:12.665   942-942   MediaPlayer             com....video_player_surface_example  V  cleanDrmObj: mDrmObj=null mDrmSessionId=null

```

</details>

### Flutter Doctor output

<details open><summary>Doctor output</summary>

```console
[✓] Flutter (Channel master, 3.23.0-14.0.pre.77, on macOS 14.1 23B2073 darwin-arm64, locale en-IN)
[✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
[✓] Xcode - develop for iOS and macOS (Xcode 15.3)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2023.3)
[✓] IntelliJ IDEA Ultimate Edition (version 2024.1.1)
[✓] Connected device (6 available)
    ! Error: Browsing on the local area network for Sanket’s iPhone. Ensure the device is unlocked and attached with a cable or associated with the same
      local area network as this Mac.
      The device must be opted into Developer Mode to connect wirelessly. (code -27)
[✓] Network resources

• No issues found!

```

</details>

**Comment by darshankawar** [comment]
Thanks for the report @Neelansh-ns 
I used both code samples you provided and ran it in S10 device using latest stable and master versions but didn't see the blank screen upon resuming app from background. See below:


https://github.com/flutter/flutter/assets/67046386/8804bdbe-8b9c-46ab-b63d-0bd4d9cb9f0a


If above doesn't represent your case, please provide a short video that shows the behavior you are seeing.

**Comment by timbotimbo** [comment]
I'm the creator of the original issue you linked.

I just tried the basic surfaceview sample with Flutter 3.22.2 on some of the affected devices:
- Samsung S20FE Android 13
- emulator Android 13
- emulator Android 14


The S20FE had no resuming issues at all.
The emulators had no issue when leaving the using the home or overview button.

## 
However on both emulators leaving using the back button/back swipe breaks the entire flutter app.
Both the surfaceview and the flutter button aren't even visible anymore, but the button still clicks.

Even fully killing the app and restarting it doesn't fix it.

I reinstalled my emulators with the latest images, and this still happens.

This video is a Pixel 8 API 34 emulator on Windows 10.
I first leave the app with a swipe up, go to settings and return to the app. (works fine)
Then I leave the app using a back swipe and return, and the app is fucked.
Last I kill the app and restart it, still broken.

https://github.com/flutter/flutter/assets/11444698/9b5558f1-bf2b-418c-9e46-f5ea5e690585


<details>
<summary>Flutter doctor -v</summary>

```
[√] Flutter (Channel stable, 3.22.2, on Microsoft Windows [Version 10.0.19045.4529], locale en-GB)
    • Flutter version 3.22.2 on channel stable at C:\Flutter\flutter_windows_3.22.0-stable\flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 761747bfc5 (4 weeks ago), 2024-06-05 22:15:13 +0200
    • Engine revision edd8546116
    • Dart version 3.4.3
    • DevTools version 2.34.3

[√] Windows Version (Installed version of Windows is version 10 or higher)

[√] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
    • Android SDK at C:\Users\XXXXXXX\AppData\Local\Android\sdk
    • Platform android-34, build-tools 34.0.0
    • Java binary at: C:\Program Files\Android\Android Studio\jbr\bin\java      
    • Java version OpenJDK Runtime Environment (build 17.0.10+0--11609105)      
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe

[√] Visual Studio - develop Windows apps (Visual Studio Community 2022 17.9.0)
    • Visual Studio at C:\Program Files\Microsoft Visual Studio\2022\Community
    • Visual Studio Community 2022 version 17.9.34607.119
    • Windows 10 SDK version 10.0.22000.0

[√] Android Studio (version 2024.1)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.10+0--11609105)

[√] VS Code (version 1.90.2)
    • VS Code at C:\Users\XXXXXXX\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.90.0

[√] Connected device (5 available)
    • SM G780G (mobile)            • RF8R8021WSR   • android-arm64  • Android 13 (API 33)
    • sdk gphone64 x86 64 (mobile) • emulator-5554 • android-x64    • Android 14 (API 34) (emulator)
    • Windows (desktop)            • windows       • windows-x64    • Microsoft Windows [Version 10.0.19045.4529]
    • Chrome (web)                 • chrome        • web-javascript • Google Chrome 126.0.6478.127
    • Edge (web)                   • edge          • web-javascript • Microsoft Edge 126.0.2592.68

[√] Network resources
    • All expected network resources are available.

• No issues found!
```
</details>

##
@Neelansh-ns  can you share a screen recording and the android version of your phone?

I'm not sure if what I just ran into is your issue or an entirely new one.

##

@darshankawar if your video is of the linked repro project, something else is broken.
You should be seeing an image with mountains. The green is a backgroundcolor that only shows up when the surfaceview is invisible.
